src/systems/geometry.js

```diff
@@ -0,0 +1,133 @@
+var geometries = require('../core/geometry').geometries;
+var registerSystem = require('../core/system').registerSystem;
+var THREE = require('../lib/three');
+
+/**
+ * System for geometry component.
+ * Handle geometry caching.
+ *
+ * @member {object} cache - Mapping of stringified component data to THREE.Geometry objects.
+ * @member {object} cacheCount - Keep track of number of entities using a geometry to
+ *         know whether to dispose on removal.
+ */
+module.exports.System = registerSystem('geometry', {
+  init: function () {
+    this.cache = {};

```

Comment:
Hm, actually, it complicates the code a little bit. Because now you have to check whether a key exists and always be creating an inner object if it doesn't. Adds bits of code everywhere. Also it's a bit more clear what the data structure looks like if separate. Is it okay to keep them separate?